### ISSUE TITLE
elf game could not run in retropie image load(game_file.elf must be executable)

### ISSUE THREAD

**Issue by shaojun** [Vegz78/McAirpos#17]
Hi,
I created another similar issue by install the `retropie` on top of latest `resbian`, repo owner provided some greate suggestions to help for fix, but for simplify my procedure, I now prefer to install the `retropie os image`, but still failed to run the elf game.

I'm using `retropie-buster-4.7.1-rpi4_400.img.gz` from official website, and install the `McAirpos` by command `curl -sL https://raw.githubusercontent.com/Vegz78/McAirpos/master/install.sh | bash -`
**I did't do any system update in whole process.**

copied into some `elf` games, run them one by one  in `retro pie`, all show black screen without any error message and then quit.

Then i quit the `retro pie` to CLI, using `~/McAirpos/McAirpos/launCharc/launCharc ~/RetroPie/roms/makecode/_gamefile.elf_` got the same sympton, all show black screen without any error message.

any idea?

**Comment by Vegz78** [comment]
No, sorry. Not except keep following up on the 2 outstanding points from your previous issue:
1. [Check executable permissions for game files](https://github.com/Vegz78/McAirpos/issues/13#issuecomment-866690857), and
2. [Upload the /tmp/McAirpos.log file right after you experience a game that exits after showing only a black screen](https://github.com/Vegz78/McAirpos/issues/16#issuecomment-877784325).

For the sake of thoroughness, I just downloaded and installed the same RetroPie image, and everything worked out of the box playing your Pizza War game with sound and controller(8bitDo) in 5 quick steps;
1. Burn and boot RetroPie 4.7.1-rpi4_400 image
2. Configure controller in RetroPie to be able to move around menus and exit RetroPie to CLI
3. Install McAirpos with the [quick installer script](https://github.com/Vegz78/McAirpos#installation)
4. Activate ssh(raspi-config), then scp games into the /home/pi/RetroPie/roms/makecode folder(checked executable permissions ok)
5.  Relaunch emulationstation and play MCA games.

Update: 
There should be no major differences between installing a RetroPie image directly and installing RetroPie on top of RPi OS, except taking care not to launch RetroPie and MCA games from within a running X-Windows session, and maybe some sound/config isssues. The readymade RetroPie image is testetd to work and supported by RetroPie, on-top installations are not. 

I run RetroPie installed on top of RPi OS on all my devices, and the only times I have encountered trouble, is with version discrepencies between RPi OS and RetroPie(for instance, updating RetroPie before updating RPi OS to the same version as in the image, where RetroPie is not downgradeable if it breaks. Important to back up before updates and major changes in such a setup).

Update2:
Both sound and image through HDMI 0 on my setup(the micro hdmi output port closest to the power input port). Using the other HDMI 1 port is known to cause various issue, e.g. among many others: https://www.raspberrypi.org/forums/viewtopic.php?t=243899
